.github/workflows/docker.yml

```diff
@@ -53,7 +54,7 @@ jobs:
 
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v1
-        if: github.event_name != 'pull_request'
+        if: github.event_name == 'push' && github.repository.full_name == 'motioneye-project/motioneye' && github.base_ref == 'refs/heads/main'
```

Comment:
Now tags aren't pushed anymore

Reply:
Only push events to our `main` branch, which is what we want?

Reply:
And push events to tags.

Reply:
Ah, this is how it works and the reason it failed before with the "tags required" error message. Then it makes more sense to check for `steps.meta.outputs.tags` instead of `main` branch. Or is there a way to check for both: That a tag exists and the it is associated with the `main` branch?

Reply:
Tags require does it says because now the event is a branch and not pull request anymore. It is configured to create a tag on the default branch with `edge`, on a pr `pr-<id>` and on a tag `x`, `x.x` and `x.x.x.`. But on an push that is not to the default branch, it does not know what tag it should create. Your last idea would be a solution.

Reply:
Ah got that, `docker/metadata-action` creates the tag meant by the error message. So what is listed below `tags` includes the condition, like that it is a PR `type=ref,event=pr`?

Generally I'd like to keep the workflow running on every push event, so that one can do pushes, also on forks, locally to any feature branch, review the checks, including whether the Docker build still works, before opening a PR. But of course Docker Hub related steps must be skipped. I'm not sure which condition works best and can be achieved easily either for explicit releases only and optionally for pushes to a testing branch => `edge` builds.

I'll not touch the Docker workflow anymore now, since I obviously lack the understanding 😅. If the above aim is not easy to achieve, feel free to revert the events/conditions. While I like tests to run on every push, this is not mandatory when it gets too complicated. The build/install/setup steps looks fine now, including some extensions to `motioneye_init` for scripts and workflows which make sense anyway.

Reply:
> Is that correct?

Sounds good. The `edge` branch does not exist yet, and probably we want to call it differently, but this is trivial to adjust then. Generally having some sorts of testing branch which is not `dev` (too frequently pushed to) sounds very reasonable.

Reply:
Okay, actually you are right. However, it shouldn't be `dev` yet, until we found out whether the `ccrisan/motioneye` namespace transfer/redirect is possible or whether we need to create an own account, hence we have Docker Hub credentials.

Reply:
Currently, the image is pushed to ghcr.io. Once the DockerHub stuff is solved we just have to add the corresponding tag to the meta action and remove the comment from the login to docker hub task. Therefore I see no problem to already start publishing to ghcr.io

Reply:
Ah, I wasn't recognising. Found it: https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-docker-registry

Okay, let's simply try it out. You are right that it helps with testing the current development as well, not only the Docker container in particular, when testers can pull a Docker container instead of needing to install motionEye natively. Then I suggest to use the `dev` branch, so it does not depend on us deciding to rename it, and we get a first image right when merging this PR.

Also these "packages" are shown on our GitHub page, below the releases, which is quite nice.